### ISSUE TITLE
feat/server-name

### ISSUE THREAD

**Issue by vdedios** [42AGV/webserv#130]
### `Nginx` behaviour in different scenarios:
- Same port, different `server_names` -> serve both resources looking at `host_name`
- Same port and same `server_names` -> serves first resource, discard followings
- Different `port`, different `server_names` -> serve both resources looking at `host_name`
- No `server_names` -> serve resources, ignores `host_name` so no matter what origin request comes from
- No `server_names`, same port -> Ignores `host_name` and serves first, discard followings
- No `server_names`, different ports -> Serves both resources ignoring `host_name`

### ...and some special cases:
- Request reaches, there are `server_names` defined but it does not match any of them -> return 404
- Several server_names in a block, names already used in previous blocks are discarded

So, basically, parser should not worry about checking format/duplication etc. of ports, addresses... This should be handled when instantiating servers and connections
